src/Sylius/Bundle/CartBundle/Resources/config/services.xml

```diff
@@ -17,33 +17,42 @@
                                http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <parameters>
-        <parameter key="sylius.context.cart.class">Sylius\Component\Cart\Context\CartContext</parameter>
-
-        <parameter key="sylius.cart_provider.default.class">Sylius\Bundle\CartBundle\Provider\CartProvider</parameter>
-
         <parameter key="sylius.listener.cart.class">Sylius\Bundle\CartBundle\EventListener\CartSubscriber</parameter>
         <parameter key="sylius.listener.cart_flash.class">Sylius\Bundle\CartBundle\EventListener\FlashSubscriber</parameter>
+        <parameter key="sylius.listener.session_cart.class">Sylius\Bundle\CartBundle\EventListener\SessionCartSubscriber</parameter>
 
         <parameter key="sylius.cart.purger.class">Sylius\Bundle\CartBundle\Purger\ExpiredCartsPurger</parameter>
     </parameters>
 
     <services>
-        <service id="sylius.context.cart" class="%sylius.context.cart.class%">
-            <argument /> <!-- Storage service name comes from config -->
+        <service id="sylius.registry.cart_context" class="Sylius\Component\Registry\PrioritizedServiceRegistry">
```

Comment:
IMO you just need to copy-paste `CompositeChannelContext` and replace `channel` with `cart` :tada: There's nothing to customize :)